src/main/java/net/sf/jsqlparser/statement/SetStatement.java

```diff
@@ -80,33 +82,39 @@ public void setName(int idx, String name) {
         values.get(idx).name = name;
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Self assignment of value (SA_SELF_ASSIGNMENT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-assignment-of-field-sa-field-self-assignment).
This method contains a self assignment of a field; e.g.
Such assignments are useless, and may indicate a logic error or typo.

Reply:
I've read code 10 times and can't see problem.
SetStatement.setName(int idx, String name)
here no name conflicts (SetStatement doen't contain fields named "idx" or "name")
then i get "values" (field of SetStatement)
and get "values" element (NameExpr) with index "idx" (param)
and set it(NameExpr) field name to parameter name

where is an expression **x=x** ? i can't see